python/ray/train/torch.py

```diff
@@ -297,55 +240,21 @@ def prepare_data_loader(
         move_to_device (bool): If set, automatically move the data
             returned by the data loader to the correct device.
     """
+    return get_session().accelerator.prepare_data_loader(
```

Comment:
There's been some changes recently since for the behavior on `get_session` (that's my fault for the delay in reviewing this).

It's now possible to call `get_session` outside the training function, and this will return None. What do we want the behavior to be in that case?  Right now this will just raise a `None type has not attribute` error. 

One thing we could do is instead of have a function in session.py that does this 
```
def get_current_accelerator():
    session = get_session()
    if session is None:
        warn_session_misuse()
         raise error
    return session.accelerator
```

This will then raise an error if any of the `train.torch.*` are used outside of the training function.

but also cc @matthewdeng for thoughts.

Reply:
This sounds good to me.

Reply:
Fixed